django-cloudlaunch/cloudlaunch/backend_plugins/app_plugin.py

```diff
@@ -61,43 +61,77 @@ def sanitise_app_config(app_config):
+        Deploy this app plugin on the supplied provider.
+
+        Perform all the necessary steps to deploy this appliance. This may
+        involve provisioning cloud resources or configuring existing host(s).
+        See the definition of each method argument as some have required
+        structure.
 
         This operation is designed to be a Celery task, and thus, can contain
         long-running operations.
 
-        @type  provider: :class:`CloudBridge.CloudProvider`
-        @param provider: Cloud provider where the supplied deployment is to be
-                         created.
+        @type  name: ``str``
+        @param name: Name of this deployment.
 
         @type  task: :class:`Task`
         @param task: A Task object, which can be used to report progress. See
                      ``tasks.Task`` for the interface details and sample
                      implementation.
 
-        @type  name: ``str``
-        @param name: Name of this deployment.
-
-        @type  cloud_config: ``dict``
-        @param cloud_config: A dict containing cloud infrastructure specific
-                             configuration for this app.
-
         @type  app_config: ``dict``
-        @param app_config: A dict containing the original, unprocessed version
-                           of the app config. The app config is a merged dict
-                           of database stored settings and user-entered
-                           settings.
-
-        @type  user_data: ``object``
-        @param user_data: An object returned by the ``process_app_config()``
-                          method which contains a validated and processed
-                          version of the ``app_config``.
+        @param app_config: A dict containing the appliance configuration. The
+                           app config is a merged dict of database stored
+                           settings and user-entered settings. In addition to
+                           the static configuration of the app, such as
+                           firewall rules or access password, this should
+                           contain a url to a host configuration playbook, if
+                           such configuration step is desired.
+
+        @type  provider_config: ``dict``
+        @param provider_config: Define the details of of the infrastructure
+                                provider where the appliance should be
+                                deployed. The following keys are supported:
+                                * ``cloud_provider``: CloudBridge object of the
+                                                      cloud provider
+                                * ``cloud_config``: A dict containing cloud
+                                                    infrastructure specific
+                                                    configuration for this app
+                                * ``cloud_user_data``: An object returned by
```

Comment:
Perhaps validated_config instead of user_data?

Reply:
Given we have `app_config` and `provider_config` as the top level parameters in this method, I guess this should then be called `validated_app_config` but why is the top level `app_config` not _validated_ then, which doesn't seem ideal. Further, all this field is ever used for is instance user data. We're also hoping to get away from using instance user data for anything but passing the config ssh key so I feel it's perhaps ok to just leave it as is.